### DISCUSSION TITLE
Rsbuild uses rspack instead of jiti to load configuration files better

### DISCUSSION THREAD

**Discussion by witsaint** [web-infra-dev/rsbuild#3163] [Ideas]
Here's why I think you can replace jiti with rspack

1: Rsbuild currently uses jiti to load configuration files, rspack is a bundle tool used by rsbuild to build web applications, and jiti itself also provides a set of compile ability, so it feels redundant。

2: jiti does not support top level await

So，We can use rspack as an entry when the configuration file extension name is ts。

The strategy is to eliminate node built-in and node_module dependencies, build code, then write it to rspack's file system, and then export it，rspack also supports these features。

I tried building my own configuration file with top level await using rspack not long ago, but the output was in esm format where default was undefined。

So I had to give up and not start issure and discuss my idea. How did the team consider this

Thanks！

**Comment by chenjiahan** [comment]
We have tried to bundle the config file with `node-bundle-require`, and it cannot handle some edge cases, especially in monorepo.

You can refer to these issues for background:

- https://github.com/web-infra-dev/rspack/issues/2237
- https://github.com/web-infra-dev/rspack/pull/3222

In the long term, I think we can use Node.js built-in [--experimental-strip-types](https://nodejs.org/en/blog/release/v22.6.0) to load TypeScript config file.

**Reply by witsaint** [reply]
Bundle-require is a little slower than jiti and is sufficient until --experimental-strip-types are stabilized.

Sorry，I have a problem, the current rsbuild sourceMap does not feel very friendly, and the actual exception cannot correspond to the location of the console。

Like this
![image](https://github.com/user-attachments/assets/50617448-8213-4d41-a794-014e52d816c6)

So  if I have this kind of inaccurate positioning, how can I improve my efficiency

The specific phenomenon can be seen in the issue I raised [issue](https://github.com/web-infra-dev/rsbuild/issues/3160)

**Reply by chenjiahan** [reply]
The key to the problem is not performance, but the fact that bundle affects the path and resolve behavior of the module.

For other problems, we can discuss them in other issues, they are outside the scope of this discussion.
